### DISCUSSION TITLE
Welcome to NMGameJam2024 Discussions!

### DISCUSSION THREAD

**Discussion by MichaelMacha** [MichaelMacha/NMGameJam2024#1] [Announcements]
@MatriarchYuzi @Classified3939 

Discussions seems like an excellent way to synchronize outside of Discord alone. I'm leaving updates on this until we're ready to submit on Sunday.

General respect rules always apply, but I for one will be visiting this page frequently and would love to hear ideas and concerns here.

**Comment by MichaelMacha** [comment]
I have a basic Vampire Survivors clone going, as you can see. That poor guy had a lot of issues on the basics, but it seems that this is very close to what I imagined it would be and we're coming along smoothly.

You'll note that the player greybox has a "BPM" field. Increasing it will increase both the tempo of the music, and the rate of attack. This is done by adjusting the `pitch_shift` field of the AudioStreamPlayer, but also adjusting the `pitch_shift` of the pitch scaler on the audio bus in the proportionate opposite direction. The difference is that adjusting the pitch scale of the stream will adjust its playback rate; but adjusting the bus will only affect pitch, so by neutralizing the pitch change we can just change BPM.

I went with a basic "Word Up" drum break for now.

**Comment by Classified3939** [comment]
It's almost certainly less work than you've managed, Michael, but I've made some progress on my main-menu-test branch. It includes some basic GUI stuff, an options menu with an audio slider and mute (though i put the mute is 'not recommended'), and a temp title that was the best i could think of at the time (i guess it's my vote for what the game title is).
The background was just me messing around with the gradient tool, but i think it actually turned out pretty neat. Super-duper stretch goal would be to animate it to the tune of the music, i suppose... probably will never get to that though.
Here's a screenshot:
![image](https://github.com/user-attachments/assets/34c93c67-7512-4dca-9571-7b838017fef5)

**Reply by MichaelMacha** [reply]
I like it.

For your menu animation, have you tried just setting up an AnimationPlayer and having it pulse on loop at 120 BPM (one pulse every half second)? You can animate virtually any property in Godot. It sounds pretty doable.

I threw the drum break together in LMMS and tossed an equalizer on it to fatten it up a bit. I think it might be cool to have the enemy synchronize its actions to a specific instrument per enemy type?

I'll whittle it down a little further this morning.

If we're doing vampires, should I go with spookier instruments? Like some kind of pipe organ funk?

**Reply by MichaelMacha** [reply]
@Classified3939 Hey, I'm popping a screen overlay UI together, as you'll see in my code after my next push. (Just for showing how many hearts you have left, etc.) It would be very handy if your UI could just be popped into it as a node with a basic process pause mechanism; but we can discuss that at 10:00. It should be simple enough.

**Comment by MichaelMacha** [comment]
Timer was complaining that it couldn't be relied on for < 0.5 second loops, which makes sense given the nature of UPS/FPS versus signaling. I went ahead and removed `AttackTimer` and replaced it with an in-code tracker on the time of the next attack.

It does look like it's synchronizing better now.

**Comment by Classified3939** [comment]
Managed to get the menu and game working together on the test branch
